### PR TITLE
chore(examples, docs): Migrate agent to alloy in docs and examples

### PR DESCRIPTION
**Pull request by alsoba13**
Migrating agent to alloy. Expect a lot of changes in urls, pointing to Alloy doc, using alloy images and config conventions.
I could test golang-pull, which worked.

Needs to be tested:
* ebpf (docker, local and kubernetes) @kolesnikovae 
* nodejs/express-pull @petethepig 
* java @alsoba13

### CODE REVIEW

File: examples/grafana-alloy-auto-instrumentation/ebpf/docker/docker-compose.yml [hunk 1]
```diff
@@ -3,22 +3,20 @@ services:
     image: grafana/pyroscope
     ports:
     - 4040:4040
-  grafana-agent:
-    image: grafana/agent:main
+  grafana-alloy:
+    image: grafana/alloy:main
```
Comment (simonswine) [inline examples/grafana-alloy-auto-instrumentation/ebpf/docker/docker-compose.yml:7]: Maybe we should use a released version for alloy too?


```suggestion
    image: grafana/alloy:latest
```

**Review by simonswine — APPROVED** [review]
LGTM. Nothing obviously wrong. But I also haven't run every example. :shipit:

**Comment by kolesnikovae** [comment]
Looks good to me! @korniltsev you probably should take a look, I suspect Alberto wanted to tag you and not me

**Comment by korniltsev** [comment]
```
➜  local git:(fa399931e) docker compose up
[+] Running 1/1
 ✘ app Error manifest for grafana/alloy:main not found: manifest unknown: manifest unknown                             3.4s
Error response from daemon: manifest for grafana/alloy:main not found: manifest unknown: manifest unknown
➜  local git:(fa399931e) pwd
/home/korniltsev/p/pyroscope/examples/grafana-alloy-auto-instrumentation/ebpf/local
```

I think we should use `grafana/alloy:latest` or `grafana/alloy-dev:latest` , there seem to be no alloy:main

**Comment by korniltsev** [comment]
The renames in ebpf/local examples breaks discovery.relabel configuration. The binary and exe name is alloy, not grafana-alloy. I propose to apply this.

```diff
diff --git a/examples/grafana-alloy-auto-instrumentation/ebpf/local/config.alloy b/examples/grafana-alloy-auto-instrumentation/ebpf/local/config.alloy
index a2f6f2d98..4c1bf1848 100644
--- a/examples/grafana-alloy-auto-instrumentation/ebpf/local/config.alloy
+++ b/examples/grafana-alloy-auto-instrumentation/ebpf/local/config.alloy
@@ -18,16 +18,16 @@ discovery.relabel "alloy" {
     // Filter needed processes
     rule {
         source_labels = ["__meta_process_exe"]
-        regex = ".*/grafana-alloy"
+        regex = ".*/alloy"
         action = "keep"
     }
      // provide arbitrary service_name label, otherwise it will be "unspecified"
     rule {
         source_labels = ["__meta_process_exe"]
         target_label = "service_name"
-        regex = ".*/grafana-alloy"
+        regex = ".*/alloy"
         action = "replace"
-        replacement = "ebpf/local/grafana-alloy"
+        replacement = "ebpf/local/alloy"
     }
 }
```

**Comment by alsoba13** [comment]
I updated the PR with the comments (latest instead of main, and fixing bin name). I didn't run the examples so I propose that everyone (dima and tolya) try and fix their examples in separate PRs.

**Pull request merged**